src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/UserController.kt

```diff
@@ -0,0 +1,26 @@
+package org.vaccineimpact.api.app.controllers
+
+import org.vaccineimpact.api.app.ActionContext
+import org.vaccineimpact.api.app.controllers.endpoints.SecuredEndpoint
+import org.vaccineimpact.api.app.errors.UnknownObjectError
+import org.vaccineimpact.api.models.User
+
+class UserController(context: ControllerContext) : AbstractController(context)
+{
+    override val urlComponent = "/users"
+    override val endpoints = listOf(
+            SecuredEndpoint("/:username/", this::getUser, emptySet())
+    )
+
+    fun getUser(context: ActionContext): User
+    {
+        var userName = userName(context)
+        var user = repos.user().use { it.getUserByUsername(userName(context))  }
+        if (user == null)
+            throw UnknownObjectError(userName, "Username")
```

Comment:
Although perhaps, since it's not clear from the repositories point of view that the endpoints are used for different things, we should bring `getUserByEmail` in line with everything else, and in `DatabasePasswordAuthenticator` catch the `UnknownObjectException` and rethrow it as a `CredentialsException`.

On other hand, I generally don't like using exceptions for logic like that. I regard the `UnknownObjectException` in a different light, because that one bubbles all the way to the top and is then handled by the general "turn this exception into an API error" mechanism.

So I leave it up to you whether to bring `getUserByEmail` in line, but either way I would like `getUserByUsername` to be like the other, ordinary repository endpoints.